File: leetcode/solutions/1967.py
# https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/

class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        res = 0

        for pattern in patterns:
            if pattern in word:
                res += 1

        return res

File: leetcode/solutions/1967.rs
// https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/

impl Solution {
    pub fn num_of_strings(patterns: Vec<String>, word: String) -> i32 {
        patterns.iter().filter(|x| word.contains(*x)).count() as i32
    }
}
